osmnx/simplification.py

```diff
@@ -245,6 +272,17 @@ def simplify_graph(G, strict=True, remove_rings=True, track_merged=False):
         topologically simplified graph, with a new `geometry` attribute on
         each simplified edge
     """
+    if strict is not None:
+        msg = (
+            "The `strict` parameter has been deprecated and will be removed in "
+            "the v2.0.0 release. Use the `endpoint_attrs` parameter instead to "
+            "relax simplification strictness. For example, `endpoint_attrs=None` "
+            "reproduces the old `strict=True` behvavior and `endpoint_attrs=['osmid']` "
+            "reproduces the old `strict=False` behavior."
```

Comment:
Maybe add this in the changelog, a migration guide and/or the PR message.

Reply:
Good idea. Added to PR comment.